include/swift/SIL/SILFunction.h

```diff
@@ -879,11 +875,20 @@ class SILFunction
 
   /// Returns true if this function can be inlined into a fragile function
   /// body.
-  bool hasValidLinkageForFragileInline() const { return isSerialized(); }
+  /// If the caller is [serialized] due to explicit `@inlinable` but
+  /// the callee was [serialized_for_package] due to package-cmo, the
+  /// callee is [serialized] and might contain loadable types, which
+  /// is normally disallowed. This check prevents such loadable types
+  /// from being inlined.
+  bool canBeSerializedIntoCaller(
```

Comment:
Lets call this predicate something like:

```
bool canBeInlinedIntoCaller(SerializedKind_t);
```

It should return `true` merely looking at the callee's and caller's serialization kind. The table we had discussed:

```
canInlineInto:
                                                     Caller
                                 not_serialized | serialized_for_pkg | serialized
        not_serialized         |      ok                no                no                         
Callee  serialized_for_pkg     |      ok                ok                no
        serialized             |      ok                ok                ok
```

Reply:
I'm not sure if we should return true if both callee and caller are not_serialized. If callee is not_serialized, we can't guarantee that it could be inlined into caller since it might contain private symbols.

Reply:
Updated as discussed.